### PR TITLE
fix: update example lockfiles

### LINKED ISSUE #4316: [turborepo] could not start a new project
### What version of Turborepo are you using?

1.8.5

### What package manager are you using / does the bug impact?

Yarn v1

### What operating system are you using?

Mac

### Describe the Bug

could not start a new project from the example, i am new to turbo and i want to setup a monorepo, two dashboards,
```
waqasali@waqass-MBP new % npx degit vercel/turbo/examples/with-yarn with-yarn
Need to install the following packages:
  degit@2.8.4
Ok to proceed? (y) y
> cloned vercel/turbo#HEAD to with-yarn
waqasali@waqass-MBP new % cd with-yarn 
waqasali@waqass-MBP with-yarn % yarn
yarn install v1.22.19
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
✨  Done in 3.84s.
waqasali@waqass-MBP with-yarn % git init
Initialized empty Git repository in /Users/waqasali/doerz/eduApply/monorepo/new/with-yarn/.git/
waqasali@waqass-MBP with-yarn % git add .
```
on yarn dev
it gives me the error
```
docs:dev: error - ../../packages/ui/Button.tsx
docs:dev: Module parse failed: Unexpected token (4:9)
docs:dev: You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
docs:dev: | 
docs:dev: | export const Button = () => {
docs:dev: >   return <button>Boop</button>;
docs:dev: | };
docs:dev: |

```

same error for another app

### other info
node version: 16.9.0
i am on mac m2

### My global installed packages
```
waqasali@waqass-MBP with-yarn % npm list -g --depth=0
/Users/waqasali/.nvm/versions/node/v16.19.0/lib
├── corepack@0.15.1
├── npm@8.19.3
├── turbo@1.8.5
└── yarn@1.22.19
```

### Expected Behavior

on yarn dev both the apps should start

### To Reproduce

run new project from example with-yarn
`npx degit vercel/turbo/examples/with-yarn with-yarn`

### Reproduction Repo

_No response_

### PR DESCRIPTION
**Pull request by chris-olszewski**
### Description

Update various lockfiles so that users who copy examples directly don't face unexpected breakages due to outdated lockfiles.

Fixes #4316 (and I believe #4307)

### Testing Instructions

As part of this PR, I changed the example tests so we run them using the lockfiles that are checked in. This should help us catch breakages due to old versions. I believe it will become increasingly important if we roll out @tknickman's new version of `create-turbo` that allows users to specify an example to use as a template as the template includes these lockfiles.

### CODE REVIEW

File: cli/integration_tests/setup_git.sh [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 git init ${TARGET_DIR} --quiet --initial-branch=main
 GIT_ARGS="--git-dir=${TARGET_DIR}/.git --work-tree=${TARGET_DIR}"
 git ${GIT_ARGS} config user.email "turbo-test@example.com"
 git ${GIT_ARGS} config user.name "Turbo Test"
 echo "script-shell=$(which bash)" > ${TARGET_DIR}/.npmrc
-npm --prefix=${TARGET_DIR} install --silent
+# Some examples don't use npm
+if [[ "$SKIP_INSTALL" != "--skip-install" ]]; then
+  npm --prefix=${TARGET_DIR} install --silent
+fi
```
Comment (mehulkar) [inline cli/integration_tests/setup_git.sh:13]: similar thing in #4190 (which i still haven't gotten passing), fyi https://github.com/vercel/turbo/pull/4190/files#diff-b7727a8fd178a7b4189de2fd392fa25a3f07140042eafbb2e339fa3407c63e01R14-R16

we can use this, it's fine

**Review by mehulkar — COMMENTED** [review]
i think this makes sense, but
- I'm not sure if keeping lockfiles in all examples is what we want. (i'm good with it, but not sure why it's not the case now)
- the test script also updates to `turbo@canary`, so you might need 2 install steps (first with lockfile, then after the turbo@canary update if there was one.

File: examples_tests/setup.sh [hunk 2]
```diff
@@ -30,7 +25,7 @@ function set_package_manager() {
 }
 
 # Set the packageManger version
-NPM_PACKAGE_MANAGER_VALUE="npm@8.1.2"
+NPM_PACKAGE_MANAGER_VALUE="npm@8.12.1"
```
Comment (chris-olszewski) [inline examples_tests/setup.sh:28]: Updating this to match the npm value that we use in `with-npm`

**Comment by chris-olszewski** [comment]
> the test script also updates to turbo@canary, so you might need 2 install steps (first with lockfile, then after the turbo@canary update if there was one.

If I'm reading `setup.sh` correctly, I think that installs only happen after we've bumped the turbo version.

**Comment by mehulkar** [comment]
> I think that installs only happen after we've bumped the turbo version.

yeah, but I think in a _safe_ npm world, you want `npm ci` to get `node_modules/` populated as "before the fact", and _then_ make updates? If you do edit package.json and then `npm i`, I don't think you would get the same result, not sure. Also very possible this is FUD, but I feel like I've been burned by this before and have since determined that the only way to get deterministic behavior is to do `npm i` on an "established" source of truth (meaning lockfile is up to date and `node_modules/` already exists)

File: examples_tests/yarn-with-npm.t [hunk 3]
```diff
@@ -1,4 +1,5 @@
-  $ . ${TESTDIR}/setup.sh with-npm yarn
+Ignore Yarn warning that it found a package-lock.json
```
Comment (mehulkar) [inline examples_tests/yarn-with-npm.t:1]: you could also delete the packagelock and do a commit like this: 
https://github.com/vercel/turbo/blob/1b383bbb9dcb5b0a203ee55d8fc6510a6123a142/cli/integration_tests/single_package/run-yarn.t#L4-L7

**Review by mehulkar — APPROVED** [review]
The `pnpm` tests are failing, but once those are resolved, I'm good with the general idea of this PR!

**Pull request merged**